test/server/listener_manager_impl_test.h

```diff
@@ -338,10 +340,27 @@ class ListenerManagerImplTest : public testing::TestWithParam<bool> {
     return manager_->addOrUpdateListener(listener, version_info, added_via_api);
   }
 
+  GenericSecretConfigProviderSharedPtr helperFindOrCreateGenericSecretProvider(
+      const envoy::config::core::v3::ConfigSource& sds_config_source,
+      const std::string& config_name,
+      Server::Configuration::TransportSocketFactoryContext& secret_provider_context,
+      Init::Manager& init_manager) {
+    std::function<void()> unregister_secret_provider = []() {};
+    GenericSecretConfigProviderSharedPtr secret_provider(sds_api_->create(
+        secret_provider_context, sds_config_source, config_name, unregister_secret_provider));
+    init_manager.add(*secret_provider->initTarget());
+    // secret_provider_context.initManager().add(*secret_provider->initTarget());
```

Comment:
if we uncomment and use `secret_provider_context` which is the server's transport socket context, same failure we will encounter that is reported in the linked issue.
Currently `init_manager` is being used which is the newly created while adding listener and thus the test passes.
IOW, regression test is guarding against the case when the factory context's init manager is also set to the server's init manager.

Reply:
Is it possible to avoid duplicating the code for initialization? An integration test is probably a better option than a unit test in listener_manager. See, for example, https://github.com/envoyproxy/envoy/blob/main/test/extensions/filters/http/oauth2/oauth_integration_test.cc. My understanding is that the existing test did not exercise delayed oauth2 filter initialization (e.g. it probably put filter into bootstrap). So if we can initialize server and then push LDS, we should be able to exercise the code path directly?

Reply:
sure